force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +273,19 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
+        //check if either date is null
+        if (d1 == null || d2 == null){
+            return null;
+        }
+        //Extract the year from each date
+        Integer year1 = d1.year();
+        Integer year2 = d2.year();
 
         // Subtract the difference in years
+        Integer yearDifference = Math.abs(year1 - year2);
 
-        return null; // Replace null with the variable you used to store the result
+        return yearDifference;
+        //I'm still confused on how/why in the initial method definition you're putting Date in an Integer variable?
```

Comment:
Is your question related to the word **Integer** in the method signature?: 
`public static Integer calculateYearDifference(Date d1, Date d2) {`
If so, Integer is the **return type**. We're returning the **number of years** between the two dates (which is an **integer**).

If that doesn't answer you question, let me know!